style_guides/js_style_guide.md

```diff
@@ -346,90 +476,67 @@ if (a == '') {
 
 And **never** use multiple ternaries together
 
-*Right:*
-
 ```js
-var foo = (a === b) ? 1 : 2;
-```
-
-*Wrong:*
+// good
+const foo = (a === b) ? 1 : 2;
 
-```js
-var foo = (a === b) ? 1 : (a === c) ? 2 : 3;
+// bad
+const foo = (a === b) ? 1 : (a === c) ? 2 : 3;
 ```
 
 ## Do not extend built-in prototypes
 
 Do not extend the prototype of native JavaScript objects. Your future self will
 be forever grateful.
 
-*Right:*
-
 ```js
-var a = [];
-if (!a.length) {
-  console.log('winning');
-}
-```
-
-*Wrong:*
-
-```js
-Array.prototype.empty = function() {
+// bad
+Array.prototype.empty = function () {
   return !this.length;
 }
-
-var a = [];
-if (a.empty()) {
-  console.log('losing');
-}
 ```
 
 ## Use descriptive conditions
 
-Any non-trivial conditions should be assigned to a descriptively named variables, broken into
-several names variables, or converted to be a function:
-
-*Right:*
+Any non-trivial conditions should be converted to functions or assigned to
+descriptively named variables.
 
 ```js
-var thing = ...;
-var isShape = thing instanceof Shape;
-var notSquare = !(thing instanceof Square);
-var largerThan10 = isShape && thing.size > 10;
-
-if (isShape && notSquare && largerThan10) {
-  console.log('some big polygon');
+// best
+function isShape(thing) {
+  return thing instanceof Shape;
+}
+function notSquare(thing) {
```

Comment:
> IMO functions shouldn't be encouraged unless you're using it more than once

functions are not there just for reusability. It's often better to break logic in clear contained sub-logical blocks (in the form of "helper functions"... to clarify the higher level logic. The example here may be an overkill (hence the alternative variable suggested I guess), but for more complex logic, it makes sense, for example, here's a higher level code that publishes the training material to all the course attendees:

```js
function generateAndSharePDF(course) {
  const attendees = course.getAttendees();
  for (let attendee on attendees) {
    const watermarkedDeck = watermark(course.trainingDeck, attendee);
    const url = upload(watermarkedDeck);
    share(attendee, url);
  }
  ...
}
```
reading the above, it's clear what the higher level logic does. You can alway drill down to details by looking at the `watermark`, `upload` and `share` functions individually. This is how you break a problem to a set of smaller problems, and the code should reflect that.